lib/msal-node/test/client/ConfidentialClientApplication.spec.ts

```diff
@@ -340,6 +343,89 @@ describe("ConfidentialClientApplication", () => {
             );
         });
 
+        describe("region is determined correctly", () => {
+            let acquireTokenByClientCredentialSpy: jest.SpyInstance;
+            let buildOauthClientConfigurationSpy: jest.SpyInstance;
+            let client: ConfidentialClientApplication;
+            let request: ClientCredentialRequest;
+            beforeEach(() => {
+                acquireTokenByClientCredentialSpy = jest.spyOn(
+                    ConfidentialClientApplication.prototype,
+                    <any>"acquireTokenByClientCredential"
+                );
+
+                buildOauthClientConfigurationSpy = jest.spyOn(
+                    ConfidentialClientApplication.prototype,
+                    <any>"buildOauthClientConfiguration"
+                );
+
+                client = new ConfidentialClientApplication(config);
+
+                request = {
+                    scopes: TEST_CONSTANTS.DEFAULT_GRAPH_SCOPE,
+                    skipCache: false,
+                };
+
+                process.env[MSAL_FORCE_REGION] = "eastus";
+            });
+
+            afterEach(() => {
+                delete process.env[MSAL_FORCE_REGION];
+            });
+
+            test("region is not passed in through the request, the MSAL_FORCE_REGION environment variable is used", async () => {
+                const authResult = (await client.acquireTokenByClientCredential(
+                    request
+                )) as AuthenticationResult;
+                expect(authResult.accessToken).toEqual(
+                    CONFIDENTIAL_CLIENT_AUTHENTICATION_RESULT.body.access_token
+                );
+                expect(acquireTokenByClientCredentialSpy).toHaveBeenCalledTimes(
+                    1
+                );
+                expect(
+                    buildOauthClientConfigurationSpy.mock.lastCall[3]
+                        .azureRegion
+                ).toEqual(process.env[MSAL_FORCE_REGION]);
+            });
+
+            test("region is passed in through the request, the MSAL_FORCE_REGION environment variable is not used", async () => {
+                const region = "westus";
+
+                const authResult = (await client.acquireTokenByClientCredential(
+                    { ...request, azureRegion: region }
+                )) as AuthenticationResult;
+                expect(authResult.accessToken).toEqual(
+                    CONFIDENTIAL_CLIENT_AUTHENTICATION_RESULT.body.access_token
+                );
+                expect(acquireTokenByClientCredentialSpy).toHaveBeenCalledTimes(
+                    1
+                );
+                expect(
+                    buildOauthClientConfigurationSpy.mock.lastCall[3]
+                        .azureRegion
+                ).toEqual(region);
+            });
+
+            test('region is not passed in through the request, the MSAL_FORCE_REGION environment variable is set to "DisableMsalForceRegion"', async () => {
```

Comment:
First 2 tests are spot on, but for the 3rd one there is confusion about how app dev configures this magic string `DisableMsalForceRegion`

The spec states: 

![image](https://github.com/user-attachments/assets/f40b9d2e-fd9e-46c4-9e16-631a46232749)

In other words, if the env variable `MSAL_FORCE_REGION=eastus` and app developer sets the region programatically (like at line 396) to `DisableMsalForceRegion` then you are to ignore the env variable.

Reply:
Based on your acceptance tests here as listed in the feature request: if the dev sets the region to "DisableMsalForceRegion", then the region should be set to undefined. Is that correct?

Reply:
Correct. See .NET PR: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4954